src/util/Spotify.js

```diff
@@ -48,53 +48,66 @@ export const Spotify = {
         Tracklist.js, cannot read property 'map' of undefined, and pointing on the
         traks of the TrackList.. I couldn't understand why.*/
     async search (term) {
-      /* let userAccessToken = this.getUserAccessToken();   I commented this out becaues it is already defined in the global of this file*/
+      
+      const topic = 'Searching Spotify\'s API.'
+
+      const mainTerm = encodeURI(term); // this converts the term passed as a parameter to a URI format
+
+      let response;
+      let JSONResponse;
+      let error;
+      let mappedTracks;
 
       if (!userAccessToken){
         console.log('No Access Token.');
-        return;
+        return ;

```

Comment:
remember yous should return a Promise from an async function

so that you can handle it where you call it, in this case return an error inside a Promise

Reply:
Done I copied the pattern from MDS.

Reply:
Since the first function is not called I left it blank. which I have a feeling is not the right way to do it but it works..

Reply:
Here I kind of "copy/paste"ed the example presented in [Promise.reject page of MDN ](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/reject#Using_the_static_Promise.reject()_method)

I tried it out and found out that it does not terminate the program but it does log the error to the console.

Reply:
It looks like that in my code:
```return Promise.reject(new Error(`${topic} No Access Token.`)).then(()=>{}, error => console.log(error));

Reply:
I'm still not sure I did good though...